Caused by: java.lang.NullPointerException: Expecting a provider value.

by default their is no value for `Social Provider` and gives

```
WARNING: Authentication encountered an error:                                                                                                                      
org.forgerock.openam.core.rest.authn.exceptions.RestAuthException: Login failure                                                                                   
        at org.forgerock.openam.core.rest.authn.trees.FailureProcessTreeResult.authFailureException(FailureProcessTreeResult.java:92)                              
        at org.forgerock.openam.core.rest.authn.trees.AuthTrees.processTree(AuthTrees.java:424)                                                                    
        at org.forgerock.openam.core.rest.authn.trees.AuthTrees.evaluateTreeAndProcessResult(AuthTrees.java:261)                                                   
        at org.forgerock.openam.core.rest.authn.trees.AuthTrees.invokeTree(AuthTrees.java:253)                                                                     
        at org.forgerock.openam.core.rest.authn.RestAuthenticationHandler.authenticate(RestAuthenticationHandler.java:222)                                         
        at org.forgerock.openam.core.rest.authn.http.AuthenticationServiceV1.authenticate(AuthenticationServiceV1.java:164)                                        
        at sun.reflect.GeneratedMethodAccessor77.invoke(Unknown Source)                                                                                            
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)                                                                   
        at java.lang.reflect.Method.invoke(Method.java:498)                                                                                                        
        at org.forgerock.openam.http.annotations.AnnotatedMethod.invoke(AnnotatedMethod.java:76)                                                                   
        at org.forgerock.openam.http.annotations.Endpoints$1.handle(Endpoints.java:64)                                                                             
        at org.forgerock.http.routing.Router.handle(Router.java:100)                                                                                               
        at org.forgerock.openam.audit.AbstractHttpAccessAuditFilter.filter(AbstractHttpAccessAuditFilter.java:59)                                                  
        at org.forgerock.http.handler.Handlers$1.handle(Handlers.java:53)                                                                                          
        at org.forgerock.http.routing.Router.handle(Router.java:100)                                                                                               
        at org.forgerock.openam.rest.RealmContextFilter.filter(RealmContextFilter.java:87)                                                                         
        at org.forgerock.http.handler.Handlers$1.handle(Handlers.java:53)                                                                                          
        at org.forgerock.http.routing.Router.handle(Router.java:100)                                                                                               
        at org.forgerock.http.routing.Router.handle(Router.java:100)                                                                                               
        at org.forgerock.openam.rest.RealmRoutingFactory$ChfRealmRouter.handle(RealmRoutingFactory.java:140)                                                       
        at org.forgerock.http.routing.Router.handle(Router.java:100)                                                                                               
        at org.forgerock.openam.rest.RealmRoutingFactory$ChfRealmRouter.handle(RealmRoutingFactory.java:140)                                                       
        at org.forgerock.http.handler.Handlers$UndescribedAsDescribableHandler.handle(Handlers.java:179)                                                           
        at org.forgerock.openam.rest.RealmRoutingFactory$HostnameFilter.filter(RealmRoutingFactory.java:117)                                                       
        at org.forgerock.http.handler.Handlers$1.handle(Handlers.java:53)                                                                                          
        at org.forgerock.http.routing.Router.handle(Router.java:100)                                                                                               
        at org.forgerock.http.routing.Router.handle(Router.java:100)                                                                                               
        at org.forgerock.openam.rest.CsrfFilter.filter(CsrfFilter.java:98)                                                                                         
        at org.forgerock.http.handler.Handlers$1.handle(Handlers.java:53)                                                                                          
        at org.forgerock.http.routing.ResourceApiVersionRoutingFilter.filter(ResourceApiVersionRoutingFilter.java:59)                                              
        at org.forgerock.http.handler.Handlers$1.handle(Handlers.java:53)                                                                                          
        at org.forgerock.caf.authentication.framework.AuthenticationFramework.grantAccess(AuthenticationFramework.java:188)                                        
        at org.forgerock.caf.authentication.framework.AuthenticationFramework.lambda$onValidateRequestSuccess$1(AuthenticationFramework.java:181)                  
        at org.forgerock.util.promise.Promises$CompletedPromise.thenAsync(Promises.java:260)                                                                       
        at org.forgerock.util.promise.Promises$CompletedPromise.thenAsync(Promises.java:249)                                                                       
        at org.forgerock.caf.authentication.framework.AuthenticationFramework.validateRequest(AuthenticationFramework.java:144)                                    
        at org.forgerock.caf.authentication.framework.AuthenticationFramework.processMessage(AuthenticationFramework.java:134)                                     
        at org.forgerock.caf.authentication.framework.AuthenticationFilter.filter(AuthenticationFilter.java:84)                                                    
        at org.forgerock.http.handler.Handlers$1.handle(Handlers.java:53)                                                                                          
        at org.forgerock.openam.http.GuiceHandler.handle(GuiceHandler.java:51)                                                                                     
        at org.forgerock.openam.http.HttpRoute$6.handle(HttpRoute.java:206)                                                                                        
        at org.forgerock.http.routing.Router.handle(Router.java:100)                                                                                               
        at org.forgerock.openam.dpro.session.ProofOfPossessionTokenFilter.filter(ProofOfPossessionTokenFilter.java:87)                                             
        at org.forgerock.http.handler.Handlers$1.handle(Handlers.java:53)                                                                                          
        at org.forgerock.http.swagger.OpenApiRequestFilter.filter(OpenApiRequestFilter.java:63)                                                                    
        at org.forgerock.http.handler.Handlers$1.handle(Handlers.java:53)                                                                                          
        at org.forgerock.openam.http.ApiDescriptorFilter.filter(ApiDescriptorFilter.java:139)                                                                      
        at org.forgerock.http.handler.Handlers$1.handle(Handlers.java:53)                                                                                          
        at org.forgerock.openam.http.OpenAMHttpApplication$1.filter(OpenAMHttpApplication.java:74)                                                                 
        at org.forgerock.http.handler.Handlers$1.handle(Handlers.java:53)                                                                                          
        at org.forgerock.http.filter.TransactionIdInboundFilter.filter(TransactionIdInboundFilter.java:86)                                                         
        at org.forgerock.http.handler.Handlers$1.handle(Handlers.java:53)                                                                                          
        at org.forgerock.http.servlet.HttpFrameworkServlet.service(HttpFrameworkServlet.java:264)                                                                  
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:742)                                                                                            
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)                                                       
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)                                                               
        at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)                                                                                  
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)                                                       
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)                                                               
        at org.forgerock.openam.services.datastore.DataStoreConsistencyFilter.lambda$doFilter$0(DataStoreConsistencyFilter.java:46)                                
        at org.forgerock.openam.service.datastore.ReentrantVolatileActionConsistencyController.safeExecute(ReentrantVolatileActionConsistencyController.java:37)   
        at org.forgerock.openam.services.datastore.DataStoreConsistencyFilter.doFilter(DataStoreConsistencyFilter.java:46)                                         
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)                                                       
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)                                                               
        at org.forgerock.openam.rest.ProtocolVersionFilter.doFilter(ProtocolVersionFilter.java:65)                                                                 
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)                                                       
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)                                                               
        at org.forgerock.openam.validation.ResponseValidationFilter.doFilter(ResponseValidationFilter.java:59)                                                     
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)                                                       
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)                                                               
        at org.forgerock.openam.headers.SetHeadersFilter.doFilter(SetHeadersFilter.java:80)                                                                        
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)                                                       
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)                                                               
        at com.sun.identity.setup.AMSetupFilter.doFilter(AMSetupFilter.java:115)                                                                                   
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)                                                       
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)                                                               
        at org.forgerock.openam.audit.context.AuditContextFilter.doFilter(AuditContextFilter.java:46)                                                              
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)                                                       
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)                                                               
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:199)                                                                     
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)                                                                      
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:493)                                                                  
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:137)                                                                           
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81)                                                                            
        at org.apache.catalina.valves.RemoteIpValve.invoke(RemoteIpValve.java:679)                                                                                 
        at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:660)                                                               
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)                                                                        
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)                                                                             
        at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:798)                                                                              
        at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)                                                                        
        at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:808)                                                                 
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1498)                                                                     
        at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)                                                                         
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)                                                                         
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)                                                                         
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)                                                                      
        at java.lang.Thread.run(Thread.java:748)                                                                                                                   
Caused by: org.forgerock.openam.auth.node.api.NodeProcessException: Failed to create node                                                                          
        at org.forgerock.openam.auth.trees.engine.NodeFactory.createNode(NodeFactory.java:131)                                                                     
        at org.forgerock.openam.auth.trees.engine.AuthTreeExecutor.process(AuthTreeExecutor.java:98)                                                               
        at org.forgerock.openam.core.rest.authn.trees.AuthTrees.processTree(AuthTrees.java:421)                                                                    
        ... 95 more                                                                                                                                                
Caused by: java.lang.NullPointerException: Expecting a provider value.                                                                                             
        at org.forgerock.util.Reject.checkNotNull(Reject.java:82)                                                                                                  
        at org.forgerock.oauth.OAuthClientConfiguration.<init>(OAuthClientConfiguration.java:36)                                                                   
        at org.forgerock.oauth.clients.oauth2.OAuth2ClientConfiguration.<init>(OAuth2ClientConfiguration.java:45)                                                  
        at org.forgerock.oauth.clients.oidc.OpenIDConnectClientConfiguration.<init>(OpenIDConnectClientConfiguration.java:55)                                      
        at org.forgerock.oauth.clients.oidc.AppleClientConfiguration.<init>(AppleClientConfiguration.java:20)                                                      
        at org.forgerock.oauth.OAuthClientConfiguration$Builder.build(OAuthClientConfiguration.java:126)                                                           
        at org.forgerock.oauth.clients.oidc.SignInWithApple.getOAuthClientConfiguration(SignInWithApple.java:290)                                                  
        at org.forgerock.oauth.clients.oidc.SignInWithApple.<init>(SignInWithApple.java:85)                                                                        
        at sun.reflect.GeneratedConstructorAccessor143.newInstance(Unknown Source)                                                                                 
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)                                                    
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)                                                                                         
        at com.google.inject.internal.DefaultConstructionProxyFactory$ReflectiveProxy.newInstance(DefaultConstructionProxyFactory.java:126)                        
        at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:114)                                                                  
        at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)                                                                   
        at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)                                                          
        at com.google.inject.internal.InjectorImpl$1.get(InjectorImpl.java:1050)                                                                                   
        at com.google.inject.assistedinject.FactoryProvider2.invoke(FactoryProvider2.java:836)                                                                     
        at org.forgerock.openam.auth.trees.engine.$Proxy118.create(Unknown Source)                                                                                 
        at org.forgerock.openam.auth.trees.engine.NodeFactory.createNode(NodeFactory.java:128)                                                                     
        ... 97 more                                                                                                                                                
```

entering a value clears this error.

Version `FRAM 6.5.1-eval`

--- a/src/main/java/org/forgerock/oauth/clients/oidc/SignInWithApple.java
+++ b/src/main/java/org/forgerock/oauth/clients/oidc/SignInWithApple.java
@@ -155,7 +155,9 @@ public class SignInWithApple extends AbstractSocialAuthLoginNode {
          * @return the provider.
          */
         @Attribute(order = 800)
-        String provider();
+        default String provider() {
+            return "Apple ID";
+        }
 
         /**
          * The authentication id key.
@@ -247,8 +249,9 @@ public class SignInWithApple extends AbstractSocialAuthLoginNode {
          * @return the issuer.
          */
         @Attribute(order = 1800)
-        String issuer();
-
+        default String issuer() {
+            return "https://appleid.apple.com";
+        }
         /**
          * The openid connect validation method.
          * @return the openid connect validation method.

--- a/src/main/java/org/forgerock/oauth/clients/oidc/SignInWithApple.java
+++ b/src/main/java/org/forgerock/oauth/clients/oidc/SignInWithApple.java
@@ -154,7 +154,7 @@ public class SignInWithApple extends AbstractSocialAuthLoginNode {
          * The provider. (useful if using IDM)
          * @return the provider.
          */
-        @Attribute(order = 800)
+        @Attribute(order = 800, validators = {RequiredValueValidator.class})
         default String provider() {
             return "Apple ID";
         }
@@ -248,10 +248,11 @@ public class SignInWithApple extends AbstractSocialAuthLoginNode {
          * The issuer. Must be specified to use mixup mitigation.
          * @return the issuer.
          */
-        @Attribute(order = 1800)
+        @Attribute(order = 1800, validators = {RequiredValueValidator.class})
         default String issuer() {
             return "https://appleid.apple.com";
         }
+        
         /**
          * The openid connect validation method.
          * @return the openid connect validation method.
@@ -260,7 +261,6 @@ public class SignInWithApple extends AbstractSocialAuthLoginNode {
         default SocialOpenIdConnectNode.OpenIDValidationMethod openIdValidationMethod() {
             return SocialOpenIdConnectNode.OpenIDValidationMethod.JWK_URL;
         }
-
 
         /**
          * The openid connect validation value.
@@ -273,7 +273,6 @@ public class SignInWithApple extends AbstractSocialAuthLoginNode {
         }
 
     }
-
 
     private static OAuthClientConfiguration getOAuthClientConfiguration(AppleConfig config) {
         AppleClientConfiguration.Builder<?, AppleClientConfiguration> builder =

